src/Microsoft.AspNet.Authentication.Facebook/FacebookAuthenticationHandler.cs

```diff
@@ -49,8 +49,20 @@ public FacebookAuthenticationHandler(HttpClient httpClient)
             return new TokenResponse(response);
         }
 
-        protected override async Task<AuthenticationTicket> GetUserInformationAsync(AuthenticationProperties properties, TokenResponse tokens)
+        protected override async Task<AuthenticationTicket> CreateTicketAsync(AuthenticationProperties properties, TokenResponse tokens)
         {
+            var ticket = await base.CreateTicketAsync(properties, tokens);
+            if (ticket == null)
+            {
+                return null;
+            }
+            
+            var identity = ticket.Principal?.Identity as ClaimsIdentity;
```

Comment:
this code is OK. AuthTicket has scenarios that need to support a null principal.